core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -333,16 +333,16 @@ class DAGScheduler(
       // (so we don't unnecessarily re-compute that data).
       val serLocs = mapOutputTracker.getSerializedMapOutputStatuses(shuffleDep.shuffleId)
       val locs = MapOutputTracker.deserializeMapStatuses(serLocs)
-      (0 until locs.length).foreach { i =>
-        if (locs(i) ne null) {
-          // locs(i) will be null if missing
-          stage.addOutputLoc(i, locs(i))
+      locs.zipWithIndex
+        .filter { case (status, _) => status != null } // null if missing
+        .foreach { case (ms, idx) =>
+          stage.addOutputLoc(idx, ms)
         }
-      }
     } else {
       // Kind of ugly: need to register RDDs with the cache and map output tracker here
       // since we can't do it in the RDD constructor because # of partitions is unknown
-      logInfo("Registering RDD " + rdd.id + " (" + rdd.getCreationSite + ")")
+      logInfo(s"Registering RDD ${rdd.id} (partitions: ${rdd.partitions.length}, " +
```

Comment:
Yes, but it was because I think it's important at this "stage". Do you think I should remove the partition-related addition?